marlowe-playground-client/src/MarloweEditor/View.purs

```diff
@@ -0,0 +1,129 @@
+module MarloweEditor.View where
+
+import Prelude hiding (div)
+import Data.Enum (toEnum, upFromIncluding)
+import Data.Lens ((^.))
+import Data.Maybe (Maybe(..), fromMaybe)
+import Effect.Aff.Class (class MonadAff)
+import Examples.Haskell.Contracts as HE
+import Halogen (ClassName(..), ComponentHTML, liftEffect)
+import Halogen.Classes (codeEditor, group)
+import Halogen.HTML (HTML, button, div, div_, option, section, select, slot, text)
+import Halogen.HTML.Events (onClick, onSelectedIndexChange)
+import Halogen.HTML.Properties (class_, classes, disabled, enabled, title)
+import Halogen.HTML.Properties as HTML
+import Halogen.Monaco (monacoComponent)
+import LocalStorage as LocalStorage
+import MainFrame.Types (ChildSlots, _marloweEditorPageSlot)
+import Marlowe.Monaco as MM
+import MarloweEditor.BottomPanel (bottomPanel)
+import MarloweEditor.Types (Action(..), State, _keybindings, _showBottomPanel, isValidContract, isValidContractWithHoles)
+import Monaco (getModel, setValue) as Monaco
+import Prim.TypeError (class Warn, Text)
+import StaticData as StaticData
+
+render ::
+  forall m.
+  MonadAff m =>
+  State ->
+  ComponentHTML Action ChildSlots m
+render state =
+  div_
+    [ section [ class_ (ClassName "code-panel") ]
+        [ div [ classes (codeEditor $ state ^. _showBottomPanel) ]
+            [ marloweEditor state ]
+        ]
+    , bottomPanel state
+    ]
+
+otherActions :: forall p. State -> HTML p Action
+otherActions state =
+  div [ classes [ group ] ]
+    [ editorOptions state
+    , viewAsBlocklyButton state
+    , sendToSimulatorButton state
+    ]
+
+sendToSimulatorButton ::
+  forall p.
+  Warn (Text "Create a custom tooltip element") =>
+  State ->
+  HTML p Action
+sendToSimulatorButton state =
+  button
+    ( [ onClick $ const $ Just SendToSimulator
+      , enabled enabled'
+      ]
+        <> disabledTooltip
+    )
+    [ text "Send To Simulator" ]
+  where
+  -- We only enable this button when the contract is valid and has no holes
+  enabled' = isValidContract state
+
+  disabledTooltip =
+    if enabled' then
+      []
+    else
+      {-
+        TODO: The title property generates a native tooltip in the browser, but it takes a couple
+        of seconds to appear, we should ask for a design and then implement a custom tooltip element.
+      -}
+      [ title "In order to send the contract to the simulator, it can't have errors nor holes"
```

Comment:
"A contract can only be sent to the simulator if it has no errors and no holes"

Reply:
changed